src/matlab2tikz.m

```diff
@@ -732,40 +747,54 @@ case guitypes()
     % This ensures we are either on the main or secondary axis
     label    = '';
     labelRef = '';
-    if ~isAxisPlotyy(m2t.currentHandles.gca)
+    if ~isAxisPlotyy(m2t.current.gca)
         return
     end
 
     % Get current label counter
-    labelNum = m2t.PlotyyLabelNum;
 
     if hasPlotyyReference(m2t,h)
         % Label the plot to later reference it. Only legend entries on the main
         % plotyy axis will have a label
-        labelNum = labelNum + 1;
-        label = sprintf('\\label{plotyyref:leg%d};\n\n', labelNum);
-        m2t.PlotyyLabelNum = labelNum;
+        [m2t, labelNum] = getNextValue(m2t, 'plotyylabel');
+        label = sprintf('\\label{%s};\n\n', plotyyLabelName(labelNum));
```

Comment:
Couldnt we just pass m2t to plotyyLabelName? therefore we could omit the assignment of labelNum